submarineswap.go

```diff
@@ -0,0 +1,265 @@
+package submarineswap
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/binary"
+	"errors"
+
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/txscript"
+	"github.com/btcsuite/btcutil"
+	"github.com/btcsuite/btcwallet/chain"
+	"github.com/btcsuite/btcwallet/waddrmgr"
+	"github.com/btcsuite/btcwallet/walletdb"
+	"github.com/lightningnetwork/lnd/channeldb"
+	"github.com/lightningnetwork/lnd/input"
+	"github.com/lightningnetwork/lnd/kvdb"
+)
+
+const (
+	defaultLockHeight      = 288
+	redeemWitnessInputSize = 1 + 1 + 73 + 1 + 32 + 1 + 100
+	refundWitnessInputSize = 1 + 1 + 73 + 1 + 0 + 1 + 100
+)
+
+var (
+	submarineBucket      = []byte("submarineTransactions")
+	wtxmgrNamespaceKey   = []byte("wtxmgr")
+	waddrmgrNamespaceKey = []byte("waddrmgr")
+)
+
+func genSubmarineSwapScript(swapperPubKey, payerPubKey, hash []byte, lockHeight int64) ([]byte, error) {
+	builder := txscript.NewScriptBuilder()
+
+	builder.AddOp(txscript.OP_HASH160)
+	builder.AddData(input.Ripemd160H(hash))
+	builder.AddOp(txscript.OP_EQUAL) // Leaves 0P1 (true) on the stack if preimage matches
+	builder.AddOp(txscript.OP_IF)
+	builder.AddData(swapperPubKey) // Path taken if preimage matches
+	builder.AddOp(txscript.OP_ELSE)
+	builder.AddInt64(lockHeight)
+	builder.AddOp(txscript.OP_CHECKSEQUENCEVERIFY)
+	builder.AddOp(txscript.OP_DROP)
+	builder.AddData(payerPubKey) // Refund back to payer
+	builder.AddOp(txscript.OP_ENDIF)
+	builder.AddOp(txscript.OP_CHECKSIG)
+
+	return builder.Script()
+}
+
+func saveSwapperSubmarineData(c *channeldb.ChannelStateDB, netID byte, hash []byte, creationHeight, lockHeight int64, swapperKey []byte, script []byte) error {
```

Comment:
_saveSwapperSubmarineData_ and _getSwapperSubmarineData_ should use postgress. Also better to put them in a separate file called store.go